hybris/bin/custom/airport/airportcore/resources/airportcore-items.xml

```diff
@@ -32,294 +32,505 @@
+                        <persistence type="property"/>
+                        <modifiers optional="false"/>
+                    </attribute>
+                    <attribute qualifier="pilotAide" type="Pilot">
+                        <persistence type="property"/>
+                        <modifiers optional="false"/>
+                    </attribute>
+                </attributes>
+            </itemtype>
+        </typegroup>
+
+
+        <!--DUTYFREE -->
+
+        <typegroup name="DutyFree">
+
+            <itemtype code="DutyFreeBaseStore" extends="BaseStore" autocreate="true" generate="true">
+                <attributes>
+                    <attribute qualifier="airport" type="Airport">
+                        <modifiers optional="false"/>
+                        <persistence type="property"/>
+                    </attribute>
+                </attributes>
+            </itemtype>
+
+            <itemtype code="DutyCosmeticProduct" extends="Product" autocreate="true" generate="true"
+                      jaloclass="it.uiip.airport.core.jalo.DutyCosmeticProduct">
+                <attributes>
+                    <attribute qualifier="bodyApplication" type="java.lang.String">
+                        <modifiers optional="false"/>
+                        <persistence type="property"/>
+                    </attribute>
+                    <attribute qualifier="ingredients" type="java.lang.String">
+                        <modifiers optional="false"/>
+                        <persistence type="property"/>
+                    </attribute>
+                    <attribute qualifier="composition" type="java.lang.String">
+                        <modifiers optional="false"/>
+                        <persistence type="property"/>
+                    </attribute>
+                </attributes>
+            </itemtype>
+
+            <itemtype code="DutyCosmeticColorVariantProduct" extends="VariantProduct" autocreate="true" generate="true"
+                      jaloclass="it.uiip.airport.core.jalo.DutyCosmeticColorVariantProduct">
+                <attributes>
+                    <attribute qualifier="color" type="SwatchColorSet">
+                        <modifiers optional="false"/>
+                        <persistence type="property"/>
+                    </attribute>
+                </attributes>
+            </itemtype>
+
+            <itemtype code="DutyCosmeticFragranceVariantProduct" extends="VariantProduct" autocreate="true"
+                      generate="true"
+                      jaloclass="it.uiip.airport.core.jalo.DutyCosmeticColorVariantProduct">
+                <attributes>
+                    <attribute qualifier="fragrance" type="java.lang.String">
+                        <modifiers optional="false"/>
+                        <persistence type="property"/>
+                    </attribute>
+                </attributes>
+            </itemtype>
+
```

Comment:
Considerate un unico livello di variant per i prodotti di tipo Cosmetic -> DutyCosmeticVariantProduct che estende VariantProduct;

In genere i diversi livelli di varianti, esempio Color e Size si usano per i capi di abbigliamento; ex: maglia (base product), maglia rossa (color variant product), maglia rossa XL (size variant product).

In questo caso ci si può fermare ad un unico livello.

Come attributi della variante considerare quelli che avete inserito per i base: ingredienti e composizione (togliendoli dal base) perchè sono specifici per il particolare prodotto